pkg/validation/internal/operatorhub.go

```diff
@@ -0,0 +1,148 @@
+	interfaces "github.com/operator-framework/api/pkg/validation/interfaces"
+)
+
+var OperatorHubValidator interfaces.Validator = interfaces.ValidatorFunc(validateOperatorHub)
+
+var validCapabilities = map[string]struct{}{
+	"Basic Install":     struct{}{},
+	"Seamless Upgrades": struct{}{},
+	"Full Lifecycle":    struct{}{},
+	"Deep Insights":     struct{}{},
+	"Auto Pilot":        struct{}{},
+}
+
+var validMediatypes = map[string]struct{}{
+	"image/gif":     struct{}{},
+	"image/jpeg":    struct{}{},
+	"image/png":     struct{}{},
+	"image/svg+xml": struct{}{},
+}
+
+var validCategories = map[string]struct{}{
+	"AI/Machine Learning":    struct{}{},
+	"Application Runtime":    struct{}{},
+	"Big Data":               struct{}{},
+	"Cloud Provider":         struct{}{},
+	"Developer Tools":        struct{}{},
+	"Database":               struct{}{},
+	"Integration & Delivery": struct{}{},
+	"Logging & Tracing":      struct{}{},
+	"Monitoring":             struct{}{},
+	"Networking":             struct{}{},
+	"OpenShift Optional":     struct{}{},
+	"Security":               struct{}{},
+	"Storage":                struct{}{},
+	"Streaming & Messaging":  struct{}{},
+}
+
+func validateOperatorHub(objs ...interface{}) (results []errors.ManifestResult) {
+	for _, obj := range objs {
+		switch v := obj.(type) {
+		case *manifests.Bundle:
+			results = append(results, validateBundleOperatorHub(v))
+		}
+	}
+	return results
+}
+
+func validateBundleOperatorHub(bundle *manifests.Bundle) errors.ManifestResult {
+	result := errors.ManifestResult{Name: bundle.Name}
+
+	if bundle == nil {
+		result.Add(errors.ErrInvalidBundle("Bundle is nil", nil))
+		return result
+	}
+
+	if bundle.CSV == nil {
+		result.Add(errors.ErrInvalidBundle("Bundle csv is nil", bundle.Name))
+		return result
+	}
+
+	errs := validateHubCSVSpec(*bundle.CSV)
+	for _, err := range errs {
+		result.Add(errors.ErrInvalidCSV(fmt.Sprintf("%s", err), bundle.CSV.GetName()))
+	}
+
+	return result
+}
+
+func validateHubCSVSpec(csv v1alpha1.ClusterServiceVersion) []error {
+	var errs []error
+
+	if csv.Spec.Provider.Name == "" {
+		errs = append(errs, fmt.Errorf("csv.Spec.Provider.Name not specified"))
+	}
+
+	for _, maintainer := range csv.Spec.Maintainers {
+		if maintainer.Name == "" || maintainer.Email == "" {
+			errs = append(errs, fmt.Errorf("csv.Spec.Maintainers elements should contain both name and email"))
+		}
+		if maintainer.Email != "" {
+			_, err := mail.ParseAddress(maintainer.Email)
+			if err != nil {
+				errs = append(errs, fmt.Errorf("csv.Spec.Maintainers email %s invalid", maintainer.Email))
+			}
+		}
+	}
+
+	for _, link := range csv.Spec.Links {
+		if link.Name == "" || link.URL == "" {
+			errs = append(errs, fmt.Errorf("csv.Spec.Links elements should contain both name and url"))
+		}
+		if link.URL != "" {
+			_, err := url.ParseRequestURI(link.URL)
+			if err != nil {
+				errs = append(errs, fmt.Errorf("csv.Spec.Links url %s invalid", link.URL))
+			}
+		}
+	}
+
```

Comment:
So no panics occur but all relevant errors are returned:
```suggestion
	if csv.GetAnnotations() == nil {
		csv.SetAnnotations(make(map[string]string))
	}

```